packages/graphql/tests/tck/tck-test-files/rfcs/top-level-unions.test.ts

```diff
@@ -0,0 +1,284 @@
+            "CALL {
+            CALL {
+            MATCH (this_Movie1:Movie)
+            RETURN this_Movie1 { __resolveType: \\"Movie\\", .id } as this_Movie1
+            }
+            RETURN collect(this_Movie1) AS Movie
+            }
+            CALL {
+            CALL {
+            MATCH (this_Series2:Series)
+            RETURN this_Series2 { __resolveType: \\"Series\\", .id } as this_Series2
+            }
+            RETURN collect(this_Series2) AS Series
+            }
+            RETURN Movie, Series"
+        `);
+
+        expect(formatParams(result.params)).toMatchInlineSnapshot(`"{}"`);
+    });
+
+    test("should query nested top level unions", async () => {
+        const typeDefs = gql`
+            type Movie {
+                id: ID!
+                actors: [Actor!]! @relationship(type: "ACTED_IN", direction: IN)
+            }
+
+            type Series {
+                id: ID!
+                episodes: [Episode!]! @relationship(type: "HAS_EPISODE", direction: IN)
+            }
+
+            type Actor {
+                id: ID!
+            }
+
+            type Episode {
+                id: ID!
+            }
+
+            union Production = Movie | Series
+        `;
+
+        const neoSchema = new Neo4jGraphQL({ typeDefs });
+
+        const query = gql`
+            query {
+                productions {
+                    ... on Movie {
+                        __typename
+                        id
+                        actors {
+                            id
+                        }
+                    }
+                    ... on Series {
+                        __typename
+                        id
+                        episodes {
+                            id
+                        }
+                    }
+                }
+            }
+        `;
+
+        const result = await translateQuery(neoSchema, query, {
+            req: createJwtRequest("secret", {}),
+        });
+
+        expect(formatCypher(result.cypher)).toMatchInlineSnapshot(`
+            "CALL {
+            CALL {
+            MATCH (this_Movie1:Movie)
+            RETURN this_Movie1 { __resolveType: \\"Movie\\", .id, actors: [ (this_Movie1)<-[:ACTED_IN]-(this_Movie1_actors:Actor)   | this_Movie1_actors { .id } ] } as this_Movie1
+            }
+            RETURN collect(this_Movie1) AS Movie
+            }
+            CALL {
+            CALL {
+            MATCH (this_Series2:Series)
+            RETURN this_Series2 { __resolveType: \\"Series\\", .id, episodes: [ (this_Series2)<-[:HAS_EPISODE]-(this_Series2_episodes:Episode)   | this_Series2_episodes { .id } ] } as this_Series2
+            }
+            RETURN collect(this_Series2) AS Series
+            }
+            RETURN Movie, Series"
+        `);
+
+        expect(formatParams(result.params)).toMatchInlineSnapshot(`"{}"`);
+    });
+
+    test("should query simple top level unions with where", async () => {
```

Comment:
**question**: In the [RFC](https://github.com/neo4j/graphql/blob/dev/docs/rfcs/rfc-009-top-level-unions.md) it is mentioned that we want to prevent over-fetching when we only return one type. Shall we add a test for that too?